frontend/src/components/material/useMaterialViewHelper.js

```diff
@@ -0,0 +1,141 @@
+import { computed, onMounted, ref } from 'vue'
+import { useRoute } from 'vue-router/composables'
+import * as XLSX from 'xlsx'
+import { slugify } from '@/plugins/slugify.js'
+import i18n from '@/plugins/i18n/index.js'
+import dayjs from '@/common/helpers/dayjs.js'
+import { apiStore } from '@/plugins/store/index.js'
+import { materialFromRoute } from '@/router.js'
+
+function generateFilename(camp, materialList) {
+  const filename = materialList
+    ? [
+        slugify(camp.name),
+        slugify(i18n.tc('components.material.useMaterialViewHelper.detail')),
+        slugify(materialList),
+      ]
+    : [
+        slugify(camp.name),
+        slugify(i18n.tc('components.material.useMaterialViewHelper.overview')),
+      ]
+  return [...filename, dayjs().format('YYMMDDHHmmss')].join('_') + '.xlsx'
+}
+
+async function getActivity(camp, materialItem) {
+  if (!materialItem.materialNode) {
+    return null
+  }
+  const root = await materialItem.materialNode().$href('root')
+  return camp
+    .activities()
+    .items.find((activity) => activity.rootContentNode()._meta.self === root)
+}
+
+async function getSheets(camp, collection, materialList) {
+  return await Promise.all(
+    collection.map(async ({ period, materialItems }) => {
+      const data = [
+        [
+          `${
+            materialList ?? i18n.tc('components.material.useMaterialViewHelper.overview')
+          }: ${period.description}`,
+        ],
+        [
+          i18n.tc('entity.materialItem.fields.quantity'),
+          i18n.tc('entity.materialItem.fields.unit'),
+          i18n.tc('entity.materialItem.fields.article'),
+          ...(!materialList ? [i18n.tc('entity.materialItem.fields.list')] : []),
+          i18n.tc('entity.materialItem.fields.reference'),
+        ],
+      ]
+      await Promise.all(
+        materialItems.items.map(async (materialItem) => {
+          let activity = await getActivity(camp, materialItem)
```

Comment:
I don't know why the linter did not complain
```suggestion
          const activity = await getActivity(camp, materialItem)
```